docs/source/topics/storage.rst

```diff
@@ -159,8 +159,8 @@ An example of an automated generated directory is ``.aiida/repository/sqlite_dos
 ``core.sqlite_zip``
 ===================
 
-The `core.sqlite_zip` is a storage plugin that is used to create export archives.
-It functions more or less identical to the `core.sqlite_dos` plugin, as it uses an SQLite database and a disk-objectstore container, except everything is bundled up in a `zip archive <https://en.wikipedia.org/wiki/ZIP_(file_format)>`_.
+The ``core.sqlite_zip`` is a storage plugin that is used to create export archives.
```

Comment:
Not really a comment on these changes, but to the section in general. Even though I understand in some cases it might be useful to set up a persistent profile to run scripts on the contents of an archive, I still think I'd first point the user to the "on the fly" approach here:

https://aiida--6222.org.readthedocs.build/projects/aiida-core/en/6222/howto/archive_profile.html

And then note that they can also set up a persistent profile, as you do already.

Reply:
Sure, we can add the method of creating a temporary profile. I am just not happy with the current method of the `create_profile` method that exists on the storage class. It shouldn't really be there tbh. Its a bit of an inversion of the hierarchy. Each profile contains a storage, among other things. So it doesn't make sense to have the storage create a profile.

The method now only exists for the zip and temp backends and is a bit of hack as well. Chris added this back when he refactored the archive functionality. I think the concept of it is nice, _but_, I am pretty sure the way it is currrenlty implemented leaves a whole bunch of functionality broken. The reason is that it simply creates `Profile` instance in memory and then loads that. It is _not_ added to the `Config` though. Not to the `config.json` (which might actually be desired) but also not to the config object in memory.

So I would be tempted to remove these methods from the `core.sqlite_temp` and `core.sqlite_zip` storage plugins and instead have something like:
```python
from aiida import create_temporary_profile, load_profile
profile = create_temporary_profile()  # this will create `core.sqlite_temp`
profile = create_temporary_profile('path/to/archive.aiida')  # creates `core.sqlite_zip`
load_profile(profile)
```
Since this approach only really makes sense for these two backends, I think we can make a dedicated function that caters just to the two with a super simple API.

Reply:
Agreed. We can discuss the exact API elsewhere, but very keen to have this. 👍